### PR TITLE
HTML in-app retain cycle issue

### PR DESCRIPTION
**Pull request by ketanshikharebsft**
created mutable dictionary for adding browser platform
Fixed memory leak issue for the html inapp

### CODE REVIEW

File: BlueShift-iOS-SDK/BlueShift.m [hunk 1]
```diff
@@ -667,12 +667,11 @@ - (void)performRequestQueue:(NSMutableDictionary *)parameters canBatchThisEvent:
     if([self validateSDKTrackingRequirements] == false) {
         return;
     }
-    
-    parameters[kBrowserPlatform] = [BlueShiftDeviceData currentDeviceData].operatingSystem;
-    
-    if (parameters != nil) {
+    NSMutableDictionary* mutableParams = [parameters mutableCopy];
+    [mutableParams setValue:[BlueShiftDeviceData currentDeviceData].operatingSystem forKey:kBrowserPlatform];
+    if (mutableParams != nil) {
```
Comment (rahulrvp) [inline BlueShift-iOS-SDK/BlueShift.m:672]: do we need this check here? you are already accessing this value in the above line

File: BlueShift-iOS-SDK/BlueShiftNotificationConstants.h [hunk 2]
```diff
@@ -33,7 +33,7 @@
 #define kNotificationCategoryViewCartIdentifier                         @"view_cart"
 #define kNotificationActionOpenCartIdentifier                           @"open_cart"
 
-#define kNotificationCategorySilentPushIdentifier                       @"silent_push"
+#define kNotificationCategorySilentPushIdentifier                       @"silent push"
```
Comment (rahulrvp) [inline BlueShift-iOS-SDK/BlueShiftNotificationConstants.h:36]: does this have something to do with keys from payload?

File: BlueShift-iOS-SDK/InApps/BlueShiftInAppNotificationManager.m [hunk 3]
```diff
@@ -693,6 +692,22 @@ - (void)presentInAppViewController:(BlueShiftNotificationViewController*)notific
...
+    if ([self inAppNotificationDisplayOnPage] == nil) {
+        return false;
+    } else if ([BlueshiftEventAnalyticsHelper isNotNilAndNotEmpty:displayOnScreen]) {
+        if(![[self inAppNotificationDisplayOnPage] isEqualToString:displayOnScreen]) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+    return true;
```
Comment (rahulrvp) [inline BlueShift-iOS-SDK/InApps/BlueShiftInAppNotificationManager.m:708]:
```
if (displayOnScreen == null || displayOnScreen == "" || [displayOnScreen] isEqualToString:[self inAppNotificationDisplayOnPage]) {
    return true;
} else {
    return false;
}
```

can we simplify the check like this? ONLY if it satisfies the cases here.

File: BlueShift-iOS-SDK/BlueShiftNotificationConstants.h (hunk 2 above)
Comment (ketanshikharebsft) [reply BlueShift-iOS-SDK/BlueShiftNotificationConstants.h:36]: yes. This key is coming from the push payload for silent push template type.

File: BlueShift-iOS-SDK/BlueShift.m (hunk 1 above)
Comment (ketanshikharebsft) [reply BlueShift-iOS-SDK/BlueShift.m:672]: Thanks for pointing this out. This `if` check is meant for parameters. I will make this change in the code.

**Review by rahulrvp — APPROVED** [review]
LGTM

**Pull request merged**